File: backend/add_sample_candidates.py
import os
from pymongo import MongoClient
from datetime import datetime
from dotenv import load_dotenv
import hashlib

load_dotenv()

# MongoDB connection
MONGO_URI = os.getenv('MONGODB_URI', "mongodb+srv://jobportal_user:jovinithin_123@jobportal.pnp4szn.mongodb.net/jobportal?retryWrites=true&w=majority&appName=Jobportal")

def hash_password(password):
    return hashlib.sha256(password.encode()).hexdigest()

try:
    client = MongoClient(MONGO_URI)
    db = client.jobportal
    
    # Sample candidates data
    candidates = [
        {
            "email": "john.smith@email.com",
            "password": hash_password("password123"),
            "userType": "jobseeker",
            "fullName": "John Smith",
            "title": "Senior React Developer",
            "location": "San Francisco, CA",
            "experience": "5+ years",
            "skills": ["React", "TypeScript", "Node.js", "AWS"],
            "salary": "$140k - $160k",
            "availability": "Available",
            "rating": 4.9,
            "phone": "+1-555-0101",
            "created_at": datetime.utcnow()
        },
        {
            "email": "sarah.anderson@email.com",
            "password": hash_password("password123"),
            "userType": "jobseeker",
            "fullName": "Sarah Anderson",
            "title": "AI Engineer",
            "location": "New York, NY",
            "experience": "7+ years",
            "skills": ["Python", "TensorFlow", "Machine Learning", "Docker"],
            "salary": "$160k - $180k",
            "availability": "Available",
            "rating": 4.8,
            "phone": "+1-555-0102",
            "created_at": datetime.utcnow()
        },
        {
            "email": "mike.chen@email.com",
            "password": hash_password("password123"),
            "userType": "jobseeker",
            "fullName": "Mike Chen",
            "title": "DevOps Engineer",
            "location": "Seattle, WA",
            "experience": "6+ years",
            "skills": ["Kubernetes", "AWS", "Terraform", "Jenkins"],
            "salary": "$130k - $150k",
            "availability": "Available in 2 weeks",
            "rating": 4.7,
            "phone": "+1-555-0103",
            "created_at": datetime.utcnow()
        },
        {
            "email": "emily.rodriguez@email.com",
            "password": hash_password("password123"),
            "userType": "jobseeker",
            "fullName": "Emily Rodriguez",
            "title": "Full Stack Developer",
            "location": "Austin, TX",
            "experience": "4+ years",
            "skills": ["Vue.js", "Python", "PostgreSQL", "Redis"],
            "salary": "$120k - $140k",
            "availability": "Available",
            "rating": 4.9,
            "phone": "+1-555-0104",
            "created_at": datetime.utcnow()
        }
    ]
    
    # Check if candidates already exist
    existing_count = db.users.count_documents({"userType": "jobseeker"})
    
    if existing_count > 0:
        print(f"Found {existing_count} existing candidates. Skipping insertion.")
    else:
        # Insert candidates
        result = db.users.insert_many(candidates)
        print(f"Successfully added {len(result.inserted_ids)} sample candidates to database!")
        
        # Verify insertion
        total_candidates = db.users.count_documents({"userType": "jobseeker"})
        print(f"Total candidates in database: {total_candidates}")
    
except Exception as e:
    print(f"Error: {e}")
finally:
    if 'client' in locals():
        client.close()

File: backend/scripts/addSampleCandidates.js
import mongoose from 'mongoose';
import bcrypt from 'bcryptjs';
import dotenv from 'dotenv';
import User from '../models/User.js';

dotenv.config();

const sampleCandidates = [
  {
    name: 'Sarah Johnson',
    email: 'sarah.johnson@email.com',
    userType: 'candidate',
    phone: '+1-555-0101',
    location: 'San Francisco, CA',
    profile: {
      skills: ['React', 'TypeScript', 'Node.js', 'AWS', 'MongoDB'],
      experience: 4,
      bio: 'Full-stack developer with 4+ years of experience building scalable web applications.'
    },
    title: 'Senior Full Stack Developer',
    salary: '$120,000 - $140,000',
    availability: 'Available',
    rating: 4.8
  },
  {
    name: 'Michael Chen',
    email: 'michael.chen@email.com',
    userType: 'candidate',
    phone: '+1-555-0102',
    location: 'Seattle, WA',
    profile: {
      skills: ['Python', 'Django', 'PostgreSQL', 'Docker', 'Kubernetes'],
      experience: 6,
      bio: 'Backend engineer specializing in scalable microservices and cloud architecture.'
    },
    title: 'Senior Backend Engineer',
    salary: '$130,000 - $150,000',
    availability: 'Available',
    rating: 4.9
  },
  {
    name: 'Emily Rodriguez',
    email: 'emily.rodriguez@email.com',
    userType: 'candidate',
    phone: '+1-555-0103',
    location: 'New York, NY',
    profile: {
      skills: ['Machine Learning', 'Python', 'TensorFlow', 'PyTorch', 'SQL'],
      experience: 3,
      bio: 'Data scientist with expertise in ML model development and deployment.'
    },
    title: 'Data Scientist',
    salary: '$110,000 - $130,000',
    availability: 'Available',
    rating: 4.7
  },
  {
    name: 'David Kim',
    email: 'david.kim@email.com',
    userType: 'candidate',
    phone: '+1-555-0104',
    location: 'Austin, TX',
    profile: {
      skills: ['Vue.js', 'JavaScript', 'CSS', 'Figma', 'UX Design'],
      experience: 5,
      bio: 'Frontend developer with strong design skills and user experience focus.'
    },
    title: 'Senior Frontend Developer',
    salary: '$100,000 - $120,000',
    availability: 'Available',
    rating: 4.6
  },
  {
    name: 'Jessica Thompson',
    email: 'jessica.thompson@email.com',
    userType: 'candidate',
    phone: '+1-555-0105',
    location: 'Remote',
    profile: {
      skills: ['DevOps', 'AWS', 'Terraform', 'Jenkins', 'Monitoring'],
      experience: 7,
      bio: 'DevOps engineer with extensive experience in cloud infrastructure and automation.'
    },
    title: 'Senior DevOps Engineer',
    salary: '$140,000 - $160,000',
    availability: 'Available',
    rating: 4.9
  },
  {
    name: 'Alex Martinez',
    email: 'alex.martinez@email.com',
    userType: 'candidate',
    phone: '+1-555-0106',
    location: 'Los Angeles, CA',
    profile: {
      skills: ['React Native', 'Swift', 'Kotlin', 'Firebase', 'GraphQL'],
      experience: 4,
      bio: 'Mobile developer with expertise in cross-platform and native app development.'
    },
    title: 'Mobile App Developer',
    salary: '$105,000 - $125,000',
    availability: 'Available',
    rating: 4.5
  },
  {
    name: 'Rachel Green',
    email: 'rachel.green@email.com',
    userType: 'candidate',
    phone: '+1-555-0107',
    location: 'Chicago, IL',
    profile: {
      skills: ['Cybersecurity', 'Penetration Testing', 'CISSP', 'Network Security'],
      experience: 8,
      bio: 'Cybersecurity specialist with extensive experience in threat assessment and security auditing.'
    },
    title: 'Senior Security Engineer',
    salary: '$150,000 - $170,000',
    availability: 'Available',
    rating: 4.8
  },
  {
    name: 'James Wilson',
    email: 'james.wilson@email.com',
    userType: 'candidate',
    phone: '+1-555-0108',
    location: 'Boston, MA',
    profile: {
      skills: ['Java', 'Spring Boot', 'Microservices', 'Apache Kafka', 'Redis'],
      experience: 6,
      bio: 'Java developer with strong background in enterprise applications and distributed systems.'
    },
    title: 'Senior Java Developer',
    salary: '$125,000 - $145,000',
    availability: 'Available',
    rating: 4.7
  }
];

async function addSampleCandidates() {
  try {
    await mongoose.connect(process.env.MONGODB_URI);
    console.log('Connected to MongoDB');

    const hashedPassword = await bcrypt.hash('candidate123', 10);

    for (const candidateData of sampleCandidates) {
      const existing = await User.findOne({ email: candidateData.email });
      if (existing) {
        console.log(`Candidate ${candidateData.email} already exists, updating...`);
        await User.updateOne({ email: candidateData.email }, {
          ...candidateData,
          password: hashedPassword
        });
      } else {
        await User.create({
          ...candidateData,
          password: hashedPassword
        });
        console.log(`Created candidate: ${candidateData.name} (${candidateData.email})`);
      }
    }

    console.log('\n✅ Sample candidates added successfully!');
    console.log('\nCandidate login credentials (password for all: candidate123):');
    sampleCandidates.forEach(candidate => {
      console.log(`- ${candidate.email} (${candidate.name})`);
    });

    process.exit(0);
  } catch (error) {
    console.error('Error:', error);
    process.exit(1);
  }
}

addSampleCandidates();